treewide: remove usage of deprecated apple_sdk frameworks

booxter opened the pull request:
They are not doing anything right now. This is in preparation for their
complete removal from the tree.

Note: this PR is a result of a mix of ugly regex (not AST) based
automation and some manual labor. For reference, the regex automation
part was hacked in: https://github.com/booxter/nix-clean-apple_sdk

Signed-off-by: Ihar Hrachyshka <ihar.hrachyshka@gmail.com>


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

booxter committed:
treewide: remove usage of deprecated apple_sdk framework stubs

They are not doing anything right now. This is in preparation for their
complete removal from the tree.

Note: several changes that affect the derivation inputs (e.g. removal of
references to stub paths in build instructions) were left out. They will
be cleaned up the next iteration and will require special care.

Note: this PR is a result of a mix of ugly regex (not AST) based
automation and some manual labor. For reference, the regex automation
part was hacked in: https://github.com/booxter/nix-clean-apple_sdk

Signed-off-by: Ihar Hrachyshka <ihar.hrachyshka@gmail.com>

emilazy commented:
(I hope to review mechanically this week. Thanks for this!)

booxter commented:
I've compared intermediates for aarch64-linux before and after this patch. The differences are:

- nixpkgs-manual and release-checks - which is expected;
- lixPackageSets.*.Security - gone (I believe also expected).

That's it. So for aarch64-linux, I believe we have a proof that this change is a no-op.

booxter commented:
```
diff -u <(sort old/paths | tr -s '[:space:]') <(sort new/paths | tr -s '[:space:]')
--- /dev/fd/11	2025-04-18 23:39:24
+++ /dev/fd/12	2025-04-18 23:39:24
@@ -28153,31 +28153,26 @@
 lixPackageSets.latest.editline.aarch64-linux dev=/nix/store/f15bzv82xxnd2jgffpqg07dsm7mvbs28-editline-1.17.1-dev;doc=/nix/store/gbnjddbwj02zzw3kpap5ixm92q2fqfp0-editline-1.17.1-doc;man=/nix/store/vs25y9v2k7s2avz1swd58s39xx2avz8z-editline-1.17.1-man;/nix/store/abi72kc7fcrk5xw02z9s16p0qz7kldlm-editline-1.17.1
 lixPackageSets.latest.lix.aarch64-linux debug=/nix/store/8r1d5ch9rh45fz4rb9argfdfbc57ch7k-lix-2.92.0-debug;dev=/nix/store/x7xp3krxyxjwsjxhmadv1cax2srqa9q3-lix-2.92.0-dev;devdoc=/nix/store/zbybmqdajw8xgyy6s022q51iaxlmzpq2-lix-2.92.0-devdoc;doc=/nix/store/bai14rk7gi8jz0wznby6r9dr5sg6f2f5-lix-2.92.0-doc;man=/nix/store/cg1ngvfiphkz254n5h329x4v86kf59hb-lix-2.92.0-man;/nix/store/01b4mlg06qn1fr12nwk8gmxprpm9m7ql-lix-2.92.0
 lixPackageSets.latest.nix-eval-jobs.aarch64-linux /nix/store/4z21ih5hf6jjnr5qzm2d855f6ys1n5rp-nix-eval-jobs-2.92.0
-lixPackageSets.latest.Security.aarch64-linux /nix/store/nc1s0l3ssnwydz50ww04r2kn05jfcw0b-Security-11.0
 lixPackageSets.lix_2_90.aws-sdk-cpp.aarch64-linux dev=/nix/store/4ljk3icv8316kxs3dwahka49fc3hiv14-aws-sdk-cpp-1.11.448-dev;/nix/store/k18q5ycvk2jlmk4zwrsil41bvvamg0h0-aws-sdk-cpp-1.11.448
 lixPackageSets.lix_2_90.boehmgc.aarch64-linux debug=/nix/store/q2b1b2ywny3sxdafvih5ccralqifslx2-boehm-gc-8.2.8-debug;dev=/nix/store/kz5h9s4d1q8v84fiyw3yc8d8w3bjja0w-boehm-gc-8.2.8-dev;doc=/nix/store/mv92ivk46kl4ij358wzxf1njkycw2fw8-boehm-gc-8.2.8-doc;/nix/store/dhw4glc4q9g5dq6cl0nswk45a9slhflx-boehm-gc-8.2.8
 lixPackageSets.lix_2_90.editline.aarch64-linux dev=/nix/store/f15bzv82xxnd2jgffpqg07dsm7mvbs28-editline-1.17.1-dev;doc=/nix/store/gbnjddbwj02zzw3kpap5ixm92q2fqfp0-editline-1.17.1-doc;man=/nix/store/vs25y9v2k7s2avz1swd58s39xx2avz8z-editline-1.17.1-man;/nix/store/abi72kc7fcrk5xw02z9s16p0qz7kldlm-editline-1.17.1
 lixPackageSets.lix_2_90.lix.aarch64-linux debug=/nix/store/f12g405w0fkq7vqw7wwdmgj0wchzdna7-lix-2.90.0-debug;dev=/nix/store/8rckz84gbn5l8zrljya6c5qbw64wl8zx-lix-2.90.0-dev;devdoc=/nix/store/490ldm1cp3r36a92cc5qq2mrsjwl9mi4-lix-2.90.0-devdoc;doc=/nix/store/cqd0ml1r0gqbx7zhfn75bkw0j9wy48c4-lix-2.90.0-doc;man=/nix/store/z2x01pbj6fgj939jzgzsr1w8pnlm13iz-lix-2.90.0-man;/nix/store/vm2s4mx7h36rkv2wl1jnm7q7di7kqmny-lix-2.90.0
 lixPackageSets.lix_2_90.nix-eval-jobs.aarch64-linux /nix/store/2fc4x8vbjhxkgkzhmdazw9l0d2h5jlpq-nix-eval-jobs-2.90.0
-lixPackageSets.lix_2_90.Security.aarch64-linux /nix/store/nc1s0l3ssnwydz50ww04r2kn05jfcw0b-Security-11.0
 lixPackageSets.lix_2_91.aws-sdk-cpp.aarch64-linux dev=/nix/store/4ljk3icv8316kxs3dwahka49fc3hiv14-aws-sdk-cpp-1.11.448-dev;/nix/store/k18q5ycvk2jlmk4zwrsil41bvvamg0h0-aws-sdk-cpp-1.11.448
 lixPackageSets.lix_2_91.boehmgc.aarch64-linux debug=/nix/store/7ahjv6a0j0a4jxdpa6y9cal4zvx3dcpb-boehm-gc-8.2.8-debug;dev=/nix/store/wywzhxks1187vwx5z5zayhs3k0iwkns5-boehm-gc-8.2.8-dev;doc=/nix/store/wyl40n01cialzhqrbqvg6nbjzp21wdyb-boehm-gc-8.2.8-doc;/nix/store/mnyf7iwcj3wz7vi03vlr9azkf30pcd1l-boehm-gc-8.2.8
 lixPackageSets.lix_2_91.editline.aarch64-linux dev=/nix/store/f15bzv82xxnd2jgffpqg07dsm7mvbs28-editline-1.17.1-dev;doc=/nix/store/gbnjddbwj02zzw3kpap5ixm92q2fqfp0-editline-1.17.1-doc;man=/nix/store/vs25y9v2k7s2avz1swd58s39xx2avz8z-editline-1.17.1-man;/nix/store/abi72kc7fcrk5xw02z9s16p0qz7kldlm-editline-1.17.1
 lixPackageSets.lix_2_91.lix.aarch64-linux debug=/nix/store/7ydmamrxrm2mlmwrwar8v4dxjjqjvd6h-lix-2.91.1-debug;dev=/nix/store/ll23j2aw1q8bj81x4bs8029xckw9cwcm-lix-2.91.1-dev;devdoc=/nix/store/p0m1a1g8q96jlj17yh8cwbr0x3q2g2yw-lix-2.91.1-devdoc;doc=/nix/store/bgjfbhq9i11jxck1kplphwl8zlgf7sfy-lix-2.91.1-doc;man=/nix/store/jv95cagxgvz6p4d35rpxbjqxd3nms0vz-lix-2.91.1-man;/nix/store/d4qny8mzqzy81v99hf2mpqkkj386akr0-lix-2.91.1
 lixPackageSets.lix_2_91.nix-eval-jobs.aarch64-linux /nix/store/zlhfh44y90161afk47fbah6pdnf5bmzr-nix-eval-jobs-2.91.0
-lixPackageSets.lix_2_91.Security.aarch64-linux /nix/store/nc1s0l3ssnwydz50ww04r2kn05jfcw0b-Security-11.0
 lixPackageSets.lix_2_92.aws-sdk-cpp.aarch64-linux dev=/nix/store/4ljk3icv8316kxs3dwahka49fc3hiv14-aws-sdk-cpp-1.11.448-dev;/nix/store/k18q5ycvk2jlmk4zwrsil41bvvamg0h0-aws-sdk-cpp-1.11.448
 lixPackageSets.lix_2_92.boehmgc.aarch64-linux debug=/nix/store/7ahjv6a0j0a4jxdpa6y9cal4zvx3dcpb-boehm-gc-8.2.8-debug;dev=/nix/store/wywzhxks1187vwx5z5zayhs3k0iwkns5-boehm-gc-8.2.8-dev;doc=/nix/store/wyl40n01cialzhqrbqvg6nbjzp21wdyb-boehm-gc-8.2.8-doc;/nix/store/mnyf7iwcj3wz7vi03vlr9azkf30pcd1l-boehm-gc-8.2.8
 lixPackageSets.lix_2_92.editline.aarch64-linux dev=/nix/store/f15bzv82xxnd2jgffpqg07dsm7mvbs28-editline-1.17.1-dev;doc=/nix/store/gbnjddbwj02zzw3kpap5ixm92q2fqfp0-editline-1.17.1-doc;man=/nix/store/vs25y9v2k7s2avz1swd58s39xx2avz8z-editline-1.17.1-man;/nix/store/abi72kc7fcrk5xw02z9s16p0qz7kldlm-editline-1.17.1
 lixPackageSets.lix_2_92.lix.aarch64-linux debug=/nix/store/8r1d5ch9rh45fz4rb9argfdfbc57ch7k-lix-2.92.0-debug;dev=/nix/store/x7xp3krxyxjwsjxhmadv1cax2srqa9q3-lix-2.92.0-dev;devdoc=/nix/store/zbybmqdajw8xgyy6s022q51iaxlmzpq2-lix-2.92.0-devdoc;doc=/nix/store/bai14rk7gi8jz0wznby6r9dr5sg6f2f5-lix-2.92.0-doc;man=/nix/store/cg1ngvfiphkz254n5h329x4v86kf59hb-lix-2.92.0-man;/nix/store/01b4mlg06qn1fr12nwk8gmxprpm9m7ql-lix-2.92.0
 lixPackageSets.lix_2_92.nix-eval-jobs.aarch64-linux /nix/store/4z21ih5hf6jjnr5qzm2d855f6ys1n5rp-nix-eval-jobs-2.92.0
-lixPackageSets.lix_2_92.Security.aarch64-linux /nix/store/nc1s0l3ssnwydz50ww04r2kn05jfcw0b-Security-11.0
 lixPackageSets.stable.aws-sdk-cpp.aarch64-linux dev=/nix/store/4ljk3icv8316kxs3dwahka49fc3hiv14-aws-sdk-cpp-1.11.448-dev;/nix/store/k18q5ycvk2jlmk4zwrsil41bvvamg0h0-aws-sdk-cpp-1.11.448
 lixPackageSets.stable.boehmgc.aarch64-linux debug=/nix/store/7ahjv6a0j0a4jxdpa6y9cal4zvx3dcpb-boehm-gc-8.2.8-debug;dev=/nix/store/wywzhxks1187vwx5z5zayhs3k0iwkns5-boehm-gc-8.2.8-dev;doc=/nix/store/wyl40n01cialzhqrbqvg6nbjzp21wdyb-boehm-gc-8.2.8-doc;/nix/store/mnyf7iwcj3wz7vi03vlr9azkf30pcd1l-boehm-gc-8.2.8
 lixPackageSets.stable.editline.aarch64-linux dev=/nix/store/f15bzv82xxnd2jgffpqg07dsm7mvbs28-editline-1.17.1-dev;doc=/nix/store/gbnjddbwj02zzw3kpap5ixm92q2fqfp0-editline-1.17.1-doc;man=/nix/store/vs25y9v2k7s2avz1swd58s39xx2avz8z-editline-1.17.1-man;/nix/store/abi72kc7fcrk5xw02z9s16p0qz7kldlm-editline-1.17.1
 lixPackageSets.stable.lix.aarch64-linux debug=/nix/store/7ydmamrxrm2mlmwrwar8v4dxjjqjvd6h-lix-2.91.1-debug;dev=/nix/store/ll23j2aw1q8bj81x4bs8029xckw9cwcm-lix-2.91.1-dev;devdoc=/nix/store/p0m1a1g8q96jlj17yh8cwbr0x3q2g2yw-lix-2.91.1-devdoc;doc=/nix/store/bgjfbhq9i11jxck1kplphwl8zlgf7sfy-lix-2.91.1-doc;man=/nix/store/jv95cagxgvz6p4d35rpxbjqxd3nms0vz-lix-2.91.1-man;/nix/store/d4qny8mzqzy81v99hf2mpqkkj386akr0-lix-2.91.1
 lixPackageSets.stable.nix-eval-jobs.aarch64-linux /nix/store/zlhfh44y90161afk47fbah6pdnf5bmzr-nix-eval-jobs-2.91.0
-lixPackageSets.stable.Security.aarch64-linux /nix/store/nc1s0l3ssnwydz50ww04r2kn05jfcw0b-Security-11.0
 lixStatic.aarch64-linux dev=/nix/store/j59lcmkjxrqxzhli4q20hmbqfzghd0b1-lix-static-aarch64-unknown-linux-musl-2.91.1-dev;/nix/store/f361qszhs0v7adcqz8srzamcpqzqh27r-lix-static-aarch64-unknown-linux-musl-2.91.1
 lizardfs.aarch64-linux /nix/store/rl131x1rx5582pn6gifbkcng3wcsb4ll-lizardfs-3.13.0-rc3
 lk-jwt-service.aarch64-linux /nix/store/5cql7s8y498gz4zj0zig0cj8cp65k5i9-lk-jwt-service-0.2.3
@@ -32025,7 +32020,7 @@
 nixpkgs-hammering.aarch64-linux /nix/store/y0lin88a3ppg5azg4s8qh67zfal47hzc-nixpkgs-hammering-0-unstable-2025-02-09
 nixpkgs-lint-community.aarch64-linux /nix/store/j042qfh1ndwhc4rr5ay4czg4akp80bav-nixpkgs-lint-0.3.0
 nixpkgs-lint.aarch64-linux /nix/store/caddlywr16yldfh7sqj5ds0xjvf6b2sd-nixpkgs-lint-1
-nixpkgs-manual.aarch64-linux /nix/store/kdihmlzn20pzz0ff5lrl1xzpqv971jzk-nixpkgs-manual
+nixpkgs-manual.aarch64-linux /nix/store/bcavhgzkxh6x7a3f8p1vs2ljhsmnv3z4-nixpkgs-manual
 nixpkgs-openjdk-updater.aarch64-linux dist=/nix/store/vc2nlw4mpksi4yph5hf0dv6q0v2j2iqh-nixpkgs-openjdk-updater-0.1.0-dist;/nix/store/sixx2fiaara4zw009i6jz0rjyak20nl7-nixpkgs-openjdk-updater-0.1.0
 nixpkgs-pytools.aarch64-linux dist=/nix/store/1hd37wzwplp5jrj1wy0wl7bmf1d3hf1x-python3.12-nixpkgs-pytools-1.3.0-dist;/nix/store/sbhwylvb8l03v5c5kzcxw1daij1fibis-python3.12-nixpkgs-pytools-1.3.0
 nixpkgs-review.aarch64-linux dist=/nix/store/s1d3rz9vp414lhhh17zrsg2ahfrk36yh-nixpkgs-review-3.2.0-dist;/nix/store/3jd5phc3i39w40wjf6ab0n8fih63xihw-nixpkgs-review-3.2.0
@@ -59047,7 +59042,7 @@
 reiserfsprogs.aarch64-linux /nix/store/kggjiwbymxsqpjssad5v6i0vdvanl8br-reiserfsprogs-3.6.27
 rekor-cli.aarch64-linux /nix/store/p6ladsai5973gbp35wbb83vqp1mv6fmy-rekor-cli-1.3.10
 rekor-server.aarch64-linux /nix/store/b1ad4hsyanxfkzcmz8rnxs5m3b1k5gv6-rekor-server-1.3.10
-release-checks /nix/store/cn8bj3jjdpd7rc09avsfzldp8dzhxgja-nixpkgs-release-checks
+release-checks /nix/store/bh2wixx86rrbqbrqw3wmmz0i39fjmxrm-nixpkgs-release-checks
 release-plz.aarch64-linux /nix/store/d7ibwazcbgz1nwc8ia6hr7nbrzf28rdi-release-plz-0.3.132
 relic.aarch64-linux /nix/store/8d8h636yq0b20m59387zdf7ahq3ij5l6-relic-8.2.0
 remake.aarch64-linux /nix/store/9ykc1pd94xwgfvvbxzs8yjm75ni81qkr-remake-4.3+dbg-1.6
```

emilazy commented:
Nice, thanks! Will try to take a proper look at this today.

emilazy commented:
Thanks so much for working on this! The vast majority of the diff looks good at a glance. I think I have `stdenv.mkDerivation` correctly filtering out Darwin stubs, so I will try to see if I can assess how close to “morally zero rebuilds” this is in a few hours.

Some notes:

1. This sets things like `libGL` to `stdenv` on Darwin. I’m not convinced that’s quite right – I think I’d prefer `apple-sdk` or `null`, though I could be convinced otherwise – but unfortunately, either of these will inevitably cause “rebuilds” from the new definition ending up in dependency lists. I’d prefer to omit that for now and we can handle it in a separate PR. The fewer manual things like that that are hard to automatically check, the easier it is to gain assurance in the process here.

1. There are a fair few stubs this doesn’t catch, like `libobjc`. Ideally this would cover everything that uses `pkgs/os-specific/darwin/apple-sdk/mk-stub.nix`. However, the remaining cases are few enough that we can handle them manually, so it’s not a blocker.

1. It makes me happy when automated treewide commits contain a command to reproduce their diff in the commit message. I remember you saying there’s manual work involved in this, though; are the manual parts cleanly separable into their own commit or PR (i.e. the automated stuff alone won’t break eval by itself), or is that not really feasible? I don’t think we need to block on this, though, as long as I can get the automated validation method giving us good results.

1. `buildInputs = [ ] ++ …Darwin stuff…;` turns into `buildInputs = [ ];`, but `buildInputs = …Darwin stuff…;` gets removed entirely. Thankfully, `stdenv.mkDerivation` normalizes this, so it doesn’t affect rebuilds, but it’d be nice if the script could automatically trim definitions that it makes unconditionally empty.
 
1. Not sure what’s going on here; I guess it relates to the split‐out `libAppleWM` PR:

   ```diff
   @@ -12602,8 +12048,7 @@ with pkgs;
          # so as not to have the newly bound xorg items already in scope,  which would
          # have created a cycle.
          overrides = lib.callPackageWith __splicedPackages ../servers/x11/xorg/overrides.nix {
   -        inherit (darwin.apple_sdk.frameworks) ApplicationServices Carbon Cocoa;
   -        inherit (darwin.apple_sdk.libs) Xplugin;
   +        inherit (darwin.apple_sdk.frameworks) ApplicationServices;
            inherit (buildPackages.darwin) bootstrap_cmds;
            udev = if stdenv.hostPlatform.isLinux then udev else null;
            libdrm = if stdenv.hostPlatform.isLinux then libdrm else null;
   ```

   It applies to `vengi-tools` as well.

1. This is a total nit, but sometimes we end up with

   ```
   foo = callPackage … {
   };
   ```

   which I don’t love, formatting‐wise. That stuff should mostly get automatically moved to `pkgs/by-name` at a later date anyway though, so it’s not a blocker. Still, perhaps it’d be easy to regex away?

booxter commented:
Thanks for the effort validating this work @emilazy!

Addressing comment point by point:

1. I've reverted libGL overrides to stdenv. We can revisit this later.
2. `libobjc` is left out of scope. There are some additional cleanups that I'd like to follow up on later related to Darwin, e.g. removing injection of libiconv for isDarwin. I understand that `libobjc` is different, and frankly I have no good reason to separate it except expediency and that I haven't realized at first when I started working on my half-assed script that I should also handle it. Consider this PR the (major) part 1 of the effort.
3. On automation: sadly, I haven't approached writing the script I used to do the bulk of the job rigorously enough to cleanly separate pieces into eval-able commits. Practically speaking, we should consider this PR all manual labor, even if a machine was used to help with it.
4. I cleaned up empty lists for buildInputs and such. There were surprisingly not a lot of them.
5. Indeed I left some packages untouched where we have separate PRs for them; to avoid some conflicts. It's intentional.
6. Cleaned up the newlines.
7. Updated the commit message and PR title.

emilazy commented:
Okay, here‘s the testing methodology. We do a horrible hack to `stdenv.mkDerivation` to ”modulo out” the stub packages:

```diff
diff --git a/pkgs/stdenv/generic/make-derivation.nix b/pkgs/stdenv/generic/make-derivation.nix
index 8ecdb68797..233f215356 100644
--- a/pkgs/stdenv/generic/make-derivation.nix
+++ b/pkgs/stdenv/generic/make-derivation.nix
@@ -18,6 +18,7 @@
     extendDerivation
     filter
     findFirst
+    genAttrs
     getDev
     head
     imap1
@@ -84,9 +85,27 @@
       # separate lines, because Nix would only show the last line of the comment.
 
       # An infinite recursion here can be caused by having the attribute names of expression `e` in `.overrideAttrs(finalAttrs: previousAttrs: e)` depend on `finalAttrs`. Only the attribute values of `e` can depend on `finalAttrs`.
-      args = rattrs (args // { inherit finalPackage overrideAttrs; });
+      args = rattrs (scrub args // { inherit finalPackage overrideAttrs; });
       #              ^^^^
 
+      scrub =
+        args:
+        args
+        // genAttrs (filter (name: args ? ${name}) [
+          "depsBuildBuild"
+          "nativeBuildInputs"
+          "depsBuildTarget"
+          "depsHostHost"
+          "buildInputs"
+          "depsTargetTarget"
+          "depsBuildBuildPropagated"
+          "propagatedNativeBuildInputs"
+          "depsBuildTargetPropagated"
+          "depsHostHostPropagated"
+          "propagatedBuildInputs"
+          "depsTargetTargetPropagated"
+        ]) (name: removeDarwinStubs args.${name});
+
       overrideAttrs = f0: makeDerivationExtensible (lib.extends (lib.toExtension f0) rattrs);
 
       finalPackage = mkDerivationSimple overrideAttrs args;
@@ -190,6 +209,21 @@
       allowedRequisites = mapNullable unsafeDerivationToUntrackedOutpath attrs.allowedRequisites;
     };
 
+  removeDarwinStubs =
+    deps:
+    filter (x: x.isDarwinCompatStub or false == false) (
+      map (
+        x:
+        if isList x then
+          let
+            filtered = removeDarwinStubs x;
+          in
+          if x != [ ] && filtered == [ ] then { isDarwinCompatStub = true; } else filtered
+        else
+          x
+      ) deps
+    );
+
   makeDerivationArgument =
 
     # `makeDerivationArgument` is responsible for the `mkDerivation` arguments that
@@ -384,7 +418,7 @@
 
         outputs = outputs';
 
-        dependencies = [
+        dependencies = map (map removeDarwinStubs) [
           [
             (map (drv: getDev drv.__spliced.buildBuild or drv) (
               checkDependencyList "depsBuildBuild" depsBuildBuild
@@ -406,7 +440,7 @@
             ))
           ]
         ];
-        propagatedDependencies = [
+        propagatedDependencies = map (map removeDarwinStubs) [
           [
             (map (drv: getDev drv.__spliced.buildBuild or drv) (
               checkDependencyList "depsBuildBuildPropagated" depsBuildBuildPropagated
```

and then compare the evaluation outpaths of the base commit + this hack against the head commit + this hack.

That uncovered the following required fixes to reduce unintentional rebuilds and avoid introducing new packaging bugs:

```diff
diff --git a/pkgs/by-name/ma/maturin/package.nix b/pkgs/by-name/ma/maturin/package.nix
index 08141f1313..c9f694e68c 100644
--- a/pkgs/by-name/ma/maturin/package.nix
+++ b/pkgs/by-name/ma/maturin/package.nix
@@ -1,8 +1,10 @@
 {
   lib,
+  stdenv,
   fetchFromGitHub,
   rustPlatform,
   fetchpatch,
+  libiconv,
   testers,
   nix-update-script,
   maturin,
@@ -33,6 +35,10 @@
     })
   ];
 
+  buildInputs = lib.optionals stdenv.hostPlatform.isDarwin [
+    libiconv
+  ];
+
   # Requires network access, fails in sandbox.
   doCheck = false;
 
diff --git a/pkgs/development/libraries/libuiohook/default.nix b/pkgs/development/libraries/libuiohook/default.nix
index b75992207d..709e75bc8e 100644
--- a/pkgs/development/libraries/libuiohook/default.nix
+++ b/pkgs/development/libraries/libuiohook/default.nix
@@ -27,8 +27,8 @@
     pkg-config
   ];
 
-  buildInputs =
-    lib.optionals (!stdenv.hostPlatform.isDarwin) [
+  buildInputs = lib.optionals (!stdenv.hostPlatform.isDarwin) (
+    [
       libX11
       libxkbcommon
       xinput
@@ -42,7 +42,8 @@
       libXtst
       libXext
       libxkbfile
-    ]);
+    ])
+  );
 
   outputs = [
     "out"
diff --git a/pkgs/development/php-packages/datadog_trace/default.nix b/pkgs/development/php-packages/datadog_trace/default.nix
index 15a1163695..22aa2565e9 100644
--- a/pkgs/development/php-packages/datadog_trace/default.nix
+++ b/pkgs/development/php-packages/datadog_trace/default.nix
@@ -35,8 +35,6 @@
     [
       cargo
       rustc
-    ]
-    ++ lib.optionals stdenv.hostPlatform.isLinux [
       rustPlatform.bindgenHook
       rustPlatform.cargoSetupHook
     ];
```

But other than that it looks good. Here‘s the outpaths diff after all is said and done:

```diff
--- result-before/outpaths.json	1970-01-01 00:00:01.000000000 +0000
+++ result-after/outpaths.json	1970-01-01 00:00:01.000000000 +0000
@@ -57359,28 +57359,28 @@
     "out": "/nix/store/gjy7agr7nfpd4inslqslaaqm4g7gpb3b-ap-reflect-0.3"
   },
   "haskellPackages.apecs-gloss.aarch64-darwin": {
-    "doc": "/nix/store/950pqv5ycm9d3wyxsy57ga1rva40hqyy-apecs-gloss-0.2.4-doc",
-    "out": "/nix/store/kxxg1vgr2rz1d5mh2f0sqal4w5nprbhc-apecs-gloss-0.2.4"
+    "doc": "/nix/store/k1awxvdndrl8z2rnv5c2qvh2f3qqnssh-apecs-gloss-0.2.4-doc",
+    "out": "/nix/store/v59f2bjmkyylk3q6sia93k3jld3kn89m-apecs-gloss-0.2.4"
   },
   "haskellPackages.apecs-gloss.x86_64-darwin": {
-    "doc": "/nix/store/cc31lfx9hwr6mhfmmzcckqp5q841hry6-apecs-gloss-0.2.4-doc",
-    "out": "/nix/store/dh0y4whmdwdbvk3l5mbmkiwk7lmrdza0-apecs-gloss-0.2.4"
+    "doc": "/nix/store/gdljhrn50kncd67vx20yq8x2dh6baa2j-apecs-gloss-0.2.4-doc",
+    "out": "/nix/store/5aq9iwm0a0wkxmb90cng66b0dkx0mwm1-apecs-gloss-0.2.4"
   },
   "haskellPackages.apecs-physics-gloss.aarch64-darwin": {
-    "doc": "/nix/store/8c69kpr66v36qj6xncw896idgacqm3ca-apecs-physics-gloss-0.1.0.0-doc",
-    "out": "/nix/store/10igmcvr5iazcj9a669gwac748dv0pni-apecs-physics-gloss-0.1.0.0"
+    "doc": "/nix/store/ih7qq3a8vsixcr7vp3nb8my3j9ywcafm-apecs-physics-gloss-0.1.0.0-doc",
+    "out": "/nix/store/4lchvq6jrsl9dq4s3v6j7gn1w3ziywb1-apecs-physics-gloss-0.1.0.0"
   },
   "haskellPackages.apecs-physics-gloss.x86_64-darwin": {
-    "doc": "/nix/store/sbflkkf1hhv043w25q20g522vgnixa5k-apecs-physics-gloss-0.1.0.0-doc",
-    "out": "/nix/store/b4dv4ggpnrxi1sx97dqlhwyv1f3rppai-apecs-physics-gloss-0.1.0.0"
+    "doc": "/nix/store/y87zdxydjbf6m92hy89cghmxxdq0bzbv-apecs-physics-gloss-0.1.0.0-doc",
+    "out": "/nix/store/90igk38rclgibkdjdkjbc2h3j6ylrnzx-apecs-physics-gloss-0.1.0.0"
   },
   "haskellPackages.apecs-physics.aarch64-darwin": {
-    "doc": "/nix/store/6j2kyc3w9ch6128wn5dmfrrp3zbri02p-apecs-physics-0.4.6-doc",
-    "out": "/nix/store/salp0wwvjbq2six55wazdv84dp00sf8q-apecs-physics-0.4.6"
+    "doc": "/nix/store/y7j24an8y8f2cpgqi7i5nfwxxvj8f7j6-apecs-physics-0.4.6-doc",
+    "out": "/nix/store/5hfmyzfrxmhcf7vpxrw39ian7g5nhams-apecs-physics-0.4.6"
   },
   "haskellPackages.apecs-physics.x86_64-darwin": {
-    "doc": "/nix/store/0qsv7w5jagy7n4nyvdpqf84qd5h8dw9h-apecs-physics-0.4.6-doc",
-    "out": "/nix/store/a7k93nalkxffpkg3gzwy2px09mxgm9my-apecs-physics-0.4.6"
+    "doc": "/nix/store/gvmjqpawkrvj891fsspsfcb914x6vs1v-apecs-physics-0.4.6-doc",
+    "out": "/nix/store/j50pw17hmav4lg8gbvk6l18sqimviyh8-apecs-physics-0.4.6"
   },
   "haskellPackages.apecs-stm.aarch64-darwin": {
     "doc": "/nix/store/k3nicirzbf3ap99dr51qsiylmc8kf9qq-apecs-stm-0.2-doc",
@@ -109549,10 +109549,10 @@
     "out": "/nix/store/8lr1j31p8mcmrwwqjj9scbmi5pfmmnz1-jama-1.2.5"
   },
   "jameica.aarch64-darwin": {
-    "out": "/nix/store/mwcykf9wvm0lcgjyighswgc9yrqkcmpw-jameica-2.10.4-487"
+    "out": "/nix/store/70qv7lm878b3pa0131y7lb16m3imf8a3-jameica-2.10.4-487"
   },
   "jameica.x86_64-darwin": {
-    "out": "/nix/store/msvn005x9mfj0143gbbyi10zjp0b0a51-jameica-2.10.4-487"
+    "out": "/nix/store/zg8479vx67g04v1lwwb84ac0pnqbr7n2-jameica-2.10.4-487"
   },
   "janet.aarch64-darwin": {
     "out": "/nix/store/mk94150fffv9qa0wv6raiark7mwygwlc-janet-1.38.0"
@@ -117204,6 +117204,12 @@
     "dev": "/nix/store/1kmc218mbjz0fdcaq4chz4cdwmn9pycx-libfreeaptx-0.2.2-dev",
     "out": "/nix/store/f089y8c568yli69rj58xd22sq5gz9gy7-libfreeaptx-0.2.2"
   },
+  "libfreefare.aarch64-darwin": {
+    "out": "/nix/store/sdrjy8mh3ml5qgq7fznfqm6jfqby6y5r-libfreefare-0.4.0"
+  },
+  "libfreefare.x86_64-darwin": {
+    "out": "/nix/store/svyn2d0y0rsi8gw2hj7wjyi80j1x836p-libfreefare-0.4.0"
+  },
   "libfsm.aarch64-darwin": {
     "dev": "/nix/store/iivikzsb22q3jkxv8ng31kiijrv9vv29-libfsm-0.1pre2987_087e3389-dev",
     "lib": "/nix/store/mhw70fhzl27kpg7x8jr9n8nbsamakycg-libfsm-0.1pre2987_087e3389-lib",
@@ -123881,12 +123887,6 @@
     "man": "/nix/store/7vnjlnz4mbb76c3ljsbparxadjs086w1-lix-2.91.1-man",
     "out": "/nix/store/zc6xbjx0xlq9sjr933vfiny0xcqc8sfh-lix-2.91.1"
   },
-  "lixPackageSets.latest.Security.aarch64-darwin": {
-    "out": "/nix/store/nw915zsclxd2dl69963knh3cv9dxdhfy-Security-11.0"
-  },
-  "lixPackageSets.latest.Security.x86_64-darwin": {
-    "out": "/nix/store/i56x39rxzmvnp1791va28fddb4gjcz5a-Security-11.0"
-  },
   "lixPackageSets.latest.aws-sdk-cpp.aarch64-darwin": {
     "dev": "/nix/store/fhr6vfi0xsv55z421knc438dhydmgi5k-aws-sdk-cpp-1.11.448-dev",
     "out": "/nix/store/azpbyq4hgb6wdzdkxibb29ncln1cpyi6-aws-sdk-cpp-1.11.448"
@@ -123937,12 +123937,6 @@
   "lixPackageSets.latest.nix-eval-jobs.x86_64-darwin": {
     "out": "/nix/store/qa30pcyk9syk1gwkpwn0pi7jnyw7dhsm-nix-eval-jobs-2.92.0"
   },
-  "lixPackageSets.lix_2_90.Security.aarch64-darwin": {
-    "out": "/nix/store/nw915zsclxd2dl69963knh3cv9dxdhfy-Security-11.0"
-  },
-  "lixPackageSets.lix_2_90.Security.x86_64-darwin": {
-    "out": "/nix/store/i56x39rxzmvnp1791va28fddb4gjcz5a-Security-11.0"
-  },
   "lixPackageSets.lix_2_90.aws-sdk-cpp.aarch64-darwin": {
     "dev": "/nix/store/fhr6vfi0xsv55z421knc438dhydmgi5k-aws-sdk-cpp-1.11.448-dev",
     "out": "/nix/store/azpbyq4hgb6wdzdkxibb29ncln1cpyi6-aws-sdk-cpp-1.11.448"
@@ -123993,12 +123987,6 @@
   "lixPackageSets.lix_2_90.nix-eval-jobs.x86_64-darwin": {
     "out": "/nix/store/3b2361f7l86hfmdzgd3zfhxnyln0i828-nix-eval-jobs-2.90.0"
   },
-  "lixPackageSets.lix_2_91.Security.aarch64-darwin": {
-    "out": "/nix/store/nw915zsclxd2dl69963knh3cv9dxdhfy-Security-11.0"
-  },
-  "lixPackageSets.lix_2_91.Security.x86_64-darwin": {
-    "out": "/nix/store/i56x39rxzmvnp1791va28fddb4gjcz5a-Security-11.0"
-  },
   "lixPackageSets.lix_2_91.aws-sdk-cpp.aarch64-darwin": {
     "dev": "/nix/store/fhr6vfi0xsv55z421knc438dhydmgi5k-aws-sdk-cpp-1.11.448-dev",
     "out": "/nix/store/azpbyq4hgb6wdzdkxibb29ncln1cpyi6-aws-sdk-cpp-1.11.448"
@@ -124049,12 +124037,6 @@
   "lixPackageSets.lix_2_91.nix-eval-jobs.x86_64-darwin": {
     "out": "/nix/store/9x0f8vqywjvv6941r4fjk1g4zkxyvaym-nix-eval-jobs-2.91.0"
   },
-  "lixPackageSets.lix_2_92.Security.aarch64-darwin": {
-    "out": "/nix/store/nw915zsclxd2dl69963knh3cv9dxdhfy-Security-11.0"
-  },
-  "lixPackageSets.lix_2_92.Security.x86_64-darwin": {
-    "out": "/nix/store/i56x39rxzmvnp1791va28fddb4gjcz5a-Security-11.0"
-  },
   "lixPackageSets.lix_2_92.aws-sdk-cpp.aarch64-darwin": {
     "dev": "/nix/store/fhr6vfi0xsv55z421knc438dhydmgi5k-aws-sdk-cpp-1.11.448-dev",
     "out": "/nix/store/azpbyq4hgb6wdzdkxibb29ncln1cpyi6-aws-sdk-cpp-1.11.448"
@@ -124105,12 +124087,6 @@
   "lixPackageSets.lix_2_92.nix-eval-jobs.x86_64-darwin": {
     "out": "/nix/store/qa30pcyk9syk1gwkpwn0pi7jnyw7dhsm-nix-eval-jobs-2.92.0"
   },
-  "lixPackageSets.stable.Security.aarch64-darwin": {
-    "out": "/nix/store/nw915zsclxd2dl69963knh3cv9dxdhfy-Security-11.0"
-  },
-  "lixPackageSets.stable.Security.x86_64-darwin": {
-    "out": "/nix/store/i56x39rxzmvnp1791va28fddb4gjcz5a-Security-11.0"
-  },
   "lixPackageSets.stable.aws-sdk-cpp.aarch64-darwin": {
     "dev": "/nix/store/fhr6vfi0xsv55z421knc438dhydmgi5k-aws-sdk-cpp-1.11.448-dev",
     "out": "/nix/store/azpbyq4hgb6wdzdkxibb29ncln1cpyi6-aws-sdk-cpp-1.11.448"
@@ -133260,10 +133236,10 @@
     "out": "/nix/store/7vr3nk9bm2rhlmyr2k6ipr9fn7rah032-luabridge-2.8"
   },
   "luaformatter.aarch64-darwin": {
-    "out": "/nix/store/zpzmmwfc1s1py81yyjfafmk9w3n087g0-luaformatter-1.3.6"
+    "out": "/nix/store/8fi5i31i0jyarkhgd955aw5fbw7lpnwv-luaformatter-1.3.6"
   },
   "luaformatter.x86_64-darwin": {
-    "out": "/nix/store/d37z0fyy5djm3z3xqa8fiyncnm8l3k64-luaformatter-1.3.6"
+    "out": "/nix/store/s3mjbmx69rapcc5cxigpkqbs0jhqag07-luaformatter-1.3.6"
   },
   "luajit.aarch64-darwin": {
     "out": "/nix/store/9vvdnw9cip644kdfg27yzmy7svfqcgaw-luajit-2.1.1741730670"
@@ -144287,10 +144263,10 @@
     "out": "/nix/store/8l5rrg3fpacngj452kii66bm11mg0z82-nixpkgs-lint-1"
   },
   "nixpkgs-manual.aarch64-darwin": {
-    "out": "/nix/store/b90yhglv2l5b19fijcsqy3s452lal6hg-nixpkgs-manual"
+    "out": "/nix/store/jvnib9n76abkid7a7afg2388yh8apcah-nixpkgs-manual"
   },
   "nixpkgs-manual.x86_64-darwin": {
-    "out": "/nix/store/jxal4ri1g6p8paib00cpslcf9jqkvlb0-nixpkgs-manual"
+    "out": "/nix/store/zam99v680pvxm5cjswwshmwsmxkgnpal-nixpkgs-manual"
   },
   "nixpkgs-openjdk-updater.aarch64-darwin": {
     "dist": "/nix/store/8k9jbsga3n1k3nfsfa2ajdqpihcrg0wn-nixpkgs-openjdk-updater-0.1.0-dist",
@@ -238312,20 +238288,20 @@
     "out": "/nix/store/bdbmygg1pnpykmkqnlxhahb4mi3jn7rq-python3.12-pygal-3.0.5"
   },
   "python312Packages.pygame-ce.aarch64-darwin": {
-    "dist": "/nix/store/kmw1ziimwlshx9nsl1cspn7yqmjgazla-python3.12-pygame-ce-2.5.3-dist",
-    "out": "/nix/store/p3rcg8xmhllm4bawl82x4kyvv71pddn3-python3.12-pygame-ce-2.5.3"
+    "dist": "/nix/store/bvy00jls0jxj4j7x0qaglj4g01r94r05-python3.12-pygame-ce-2.5.3-dist",
+    "out": "/nix/store/cd5vzf646hl1i3s5hc678cn6i73xcg3r-python3.12-pygame-ce-2.5.3"
   },
   "python312Packages.pygame-ce.x86_64-darwin": {
-    "dist": "/nix/store/z06snmg20jkjgwwyhyz3swlf5jrj6fm8-python3.12-pygame-ce-2.5.3-dist",
-    "out": "/nix/store/hppxbqd9n6hgjnv48xhssrigqp9sj3sq-python3.12-pygame-ce-2.5.3"
+    "dist": "/nix/store/iibasdjm218na01p715m33wnrplhskpq-python3.12-pygame-ce-2.5.3-dist",
+    "out": "/nix/store/5gbj41v464xf9vjini16a461h3s90al0-python3.12-pygame-ce-2.5.3"
   },
   "python312Packages.pygame-gui.aarch64-darwin": {
-    "dist": "/nix/store/w9vrfv8n83hm2wbczrdmg3xggy06i69b-python3.12-pygame-gui-0613-dist",
-    "out": "/nix/store/9wvd11x2dkj6y3v6z1i5zz250mwf6sxz-python3.12-pygame-gui-0613"
+    "dist": "/nix/store/hykwvjpjpaijwkgdfhc76v3cxz1s0ary-python3.12-pygame-gui-0613-dist",
+    "out": "/nix/store/1ng0b26crj03p4mgf33j1a3g638q296i-python3.12-pygame-gui-0613"
   },
   "python312Packages.pygame-gui.x86_64-darwin": {
-    "dist": "/nix/store/9qxhqf4hz6a2nd2vhs4ms7mx8nphpprl-python3.12-pygame-gui-0613-dist",
-    "out": "/nix/store/q21x25w6h7bdfxv6wjklhl3n7n5qpgy6-python3.12-pygame-gui-0613"
+    "dist": "/nix/store/8rqd39acppdfhzv8wbxk7xky08cxri5f-python3.12-pygame-gui-0613-dist",
+    "out": "/nix/store/6k8k14k1b3wljfm0b4a7a6622ydskzk9-python3.12-pygame-gui-0613"
   },
   "python312Packages.pygame-sdl2.aarch64-darwin": {
     "dist": "/nix/store/fsvsjglhd69nzmjhl1pd9z7xcnmjwcz5-python3.12-pygame-sdl2-8.3.7.25031702-dist",
@@ -308273,20 +308249,20 @@
     "out": "/nix/store/7w6sqnpljl2rvhs3m5c7cfmdgyl0a8b4-python3.13-pygal-3.0.5"
   },
   "python313Packages.pygame-ce.aarch64-darwin": {
-    "dist": "/nix/store/sgbnysxlng611j230y2mi6cqjbrkckj4-python3.13-pygame-ce-2.5.3-dist",
-    "out": "/nix/store/7g15gmgywgh6i3xm40yigaspm8wff340-python3.13-pygame-ce-2.5.3"
+    "dist": "/nix/store/4s44nbvml7ynvwh0dlh3p5kb2fj534b7-python3.13-pygame-ce-2.5.3-dist",
+    "out": "/nix/store/dsrwqwa99za1sd5kbwd36xh3vjdvd601-python3.13-pygame-ce-2.5.3"
   },
   "python313Packages.pygame-ce.x86_64-darwin": {
-    "dist": "/nix/store/037wqavarljcxlf1avav95pq38zgg034-python3.13-pygame-ce-2.5.3-dist",
-    "out": "/nix/store/h0gqylr4f7ln9v2hl231kad8idgsps18-python3.13-pygame-ce-2.5.3"
+    "dist": "/nix/store/4jvjkcclbg9z8x7l2vywr2jnb1alq0qj-python3.13-pygame-ce-2.5.3-dist",
+    "out": "/nix/store/vn3m6fp9rs8gxh4y9icqqdh9xghrcnia-python3.13-pygame-ce-2.5.3"
   },
   "python313Packages.pygame-gui.aarch64-darwin": {
-    "dist": "/nix/store/i9y4qpiix7a08128y379fvzm1pbj44fi-python3.13-pygame-gui-0613-dist",
-    "out": "/nix/store/ky8yr7f8m08rzdm9dn1bdjayww8xy0i0-python3.13-pygame-gui-0613"
+    "dist": "/nix/store/ma4s3qnjapc0slqn696552dlmm3ws605-python3.13-pygame-gui-0613-dist",
+    "out": "/nix/store/l4kwi73rp0h7544z7yr3a0fgplwi6gif-python3.13-pygame-gui-0613"
   },
   "python313Packages.pygame-gui.x86_64-darwin": {
-    "dist": "/nix/store/0aplnhsc9i3fihhzplchlaqr08jj5y1b-python3.13-pygame-gui-0613-dist",
-    "out": "/nix/store/if1nga867lrrd8gns7d3bpxds1qz9qm9-python3.13-pygame-gui-0613"
+    "dist": "/nix/store/46dw9nrv17qq1hgqww775c85am91810v-python3.13-pygame-gui-0613-dist",
+    "out": "/nix/store/drkfxwc3gd8agx3rpilq4972jvwqq05g-python3.13-pygame-gui-0613"
   },
   "python313Packages.pygame-sdl2.aarch64-darwin": {
     "dist": "/nix/store/c58m2b7gn9wa6z34hmpnv63fa96fdlk8-python3.13-pygame-sdl2-8.3.7.25031702-dist",
@@ -338530,7 +338506,7 @@
     "out": "/nix/store/f674n4sq8jgwzk7sipcqbq1gysgjsiav-rekor-server-1.3.10"
   },
   "release-checks": {
-    "out": "/nix/store/8p62z1w2nhn9kfwvli8kv8y58p35xrkz-nixpkgs-release-checks"
+    "out": "/nix/store/izym3702akkiwlkrsk2zx6hcnry2cjr4-nixpkgs-release-checks"
   },
   "release-plz.aarch64-darwin": {
     "out": "/nix/store/csfixwwp75j277mid9q3wmss6n3ks34c-release-plz-0.3.132"
```

The Haskell stuff is because

```diff
-    apecs-physics = addPkgconfigDepends [
-      darwin.apple_sdk.frameworks.ApplicationServices
-    ] super.apecs-physics;
```

drops a `pkg-config` dependency that shouldn‘t have been there in the first place.

`libfreefare` was broken before due to using `darwin.apple_sdk` as a package and is fixed by the PR.

`jameica` and `python3Packages.pygame-ce` do terrifying things with `rec` and `buildInputs` that my hack can’t scrub; `python3Packages.pygame-gui` is downstream of that.

`Security` disappearing from the Lix package set is expected, as are changes to the manual and release checks hashes.

`luaformatter` is some strange thing where my hack is leaving a space at the start of `buildInputs` when `CoreFoundation` is present in the `antlr4_9.cpp` derivation. Doesn’t seem worth the time to investigate why given the general success of this approach.

Once you squash my fixes in I’m happy to merge this (after comparing the evaluation results to make sure nothing new comes up, if this needs rebasing before then).

emilazy commented:
Thanks again!

I’ll look at the pending split‐out PRs tomorrow, then clean up any references to stubs this didn’t handle and move them to the aliases file with a warning. That should have us set for 25.05.

This is probably going to be a bit awkward with `master` → `staging` merges, but I’ll try to handle those as they come up.
